### PR TITLE
feature: Use shell to check node cgroup driver before kubeadm init and join

### LINKED ISSUE #489: using systemd as default cgroup driver
https://kubernetes.io/docs/setup/production-environment/container-runtimes/

config docker cgroup driver:
```
cat <<EOF | sudo tee /etc/docker/daemon.json
{
  "exec-opts": ["native.cgroupdriver=systemd"],
  "log-driver": "json-file",
  "log-opts": {
    "max-size": "100m"
  },
  "storage-driver": "overlay2"
}
EOF
```

kubeadm config:
```
# kubeadm-config.yaml
kind: ClusterConfiguration
apiVersion: kubeadm.k8s.io/v1beta2
kubernetesVersion: v1.21.0
---
kind: KubeletConfiguration
apiVersion: kubelet.config.k8s.io/v1beta1
cgroupDriver: systemd
```

### PR DESCRIPTION
**Pull request by oldthreefeng**
Signed-off-by: oldthreefeng <louisehong4168@gmail.com>

### Describe what this PR does / why we need it

if we dont check node cgroupdriver and don't specific cgroup driver.  kubeadm will set it by default value.

### Does this pull request fix one issue?

Fixes #489 

### Describe how you did it

use shell to get .

### CODE REVIEW

File: runtime/kubeadm.go [hunk 1]
```diff
@@ -59,21 +60,37 @@ func (d *Default) templateFromContent(templateContent string) ([]byte, error) {
...
 	// kubernetes gt 1.20, use Containerd instead of docker
-	if VersionCompare(d.Metadata.Version, V1200) {
-		sb.Write([]byte(InitTemplateTextV1bate2))
-	} else {
-		sb.Write([]byte(InitTemplateTextV1beta1))
+	case VersionCompare(d.Metadata.Version, V1200):
+		d.KubeadmAPI = KubeadmV1beta2
+		d.CriSocket = DefaultContainerdCRISocket
+	default:
+		// 兼容一下 1.14 以下版本.
```
Comment (fanux) [inline runtime/kubeadm.go:85]: using english

**Comment by fanux** [comment]
/need test
@bxy4543

File: runtime/kubeadm.go (hunk 1 above)
Comment (oldthreefeng) [reply runtime/kubeadm.go:85]: ok.

**Review by fanux — APPROVED** [review]

**Pull request merged**